src/agent/WMICommunication.cpp

```diff
@@ -0,0 +1,204 @@
+#include "WMICommunication.h"
+#include "WMICommunication.h"
+#include "WMICommunication.h"
+
+#define _WIN32_DCOM
+#include <iostream>
+#include <comdef.h>
+#pragma comment(lib, "wbemuuid.lib")
+
+
+WMICommunication::~WMICommunication()
+{
+    initialLocatorToWMI->Release();
+    services->Release();
+    pEnumerator->Release();
+    CoUninitialize();
+}
+
+bool WMICommunication::WMIInit()
```

Comment:
I think this method can be reorganized a bit.
Problems I see here:
- each failed step calls de-initialization  on previous steps. This is error prone.
- there are throws which noone catches.

possible solution addressing both problems:
put all initialization steps into a try {} block and in catch {} call deinit on each pointer which is not null (was set).

Reply:
I agree.